### PR TITLE
Run unison as a non-root user (specifying the UID and GUID to own the files)

### PR DESCRIPTION
**Pull request by davide**
Useful for ensuring that file ownership in shared volumes matches across containers.

### CODE REVIEW

**Comment by cweagans** [comment]
I'm not opposed to this, but two changes please:

1. All of the stuff you're doing in the Dockerfile needs to live in sync.sh instead. This way, it's a configurable value without having to rebuild the container (i.e. this should be startup configuration, not build configuration).

2. Please default to root so that we're not changing behavior for existing users. I know it's not ideal, but I don't want to break existing setups.

**Comment by davide** [comment]
Good input! I added a commit with the suggested changes. Let me know if if needs further improvements.

File: sync.sh [hunk 1]
```diff
@@ -145,9 +163,15 @@ ignore = Name *___jb_tmp___*
...
 $SYNC_EXTRA_UNISON_PROFILE_OPTS
 
-" > /root/.unison/default.prf
+" > ${HOME}/.unison/default.prf
+
+# Fix permissions
+log_heading "Override folder permissions."
+
+chown -R ${UNISON_USER}:${UNISON_GROUP} $SYNC_SOURCE
+chown -R ${UNISON_USER}:${UNISON_GROUP} $SYNC_DESTINATION
```
Comment (cweagans) [inline sync.sh]: Does this affect files on the host?

Comment (davide) [inline sync.sh]: Yes, when $SYNC_SOURCE or $SYNC_DESTINATION are host mounted volumes.
This has the potential to be a very slow operation and this behavior can impact existing bg-sync users. I'll rollback this change.

**Comment by cweagans** [comment]
This looks great. Thanks for the revisions!

**Comment by davide** [comment]
You're welcome! I noticed that you already pushed a new Docker Hub based on this. Thanks!!

**Pull request merged**